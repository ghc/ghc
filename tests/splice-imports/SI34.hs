module SI34 where

-- Compiling SI34 @ R, requires SI34M2 @ R, which requires SI34M1 @ R,
-- but NOT SI34M1 @ C or SI34M2 @ C due to ImplicitStagePersistence + TemplateHaskellQuotes
import SI34M2

-- Uses the MkT constructor indirectly through SI34M2.makeMkT
foo = makeMkT 42

-- Uses the wrapper type from SI34M2
bar = wrapT (makeMkT 100)