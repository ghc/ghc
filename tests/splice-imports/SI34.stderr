[1 of 3] Compiling SI34M1           ( SI34M1.hs, nothing )
[2 of 3] Compiling SI34M2           ( SI34M2.hs, nothing )
[3 of 3] Compiling SI34             ( SI34.hs, nothing )
